.eslintrc.js

```diff
@@ -0,0 +1,263 @@
+      "@typescript-eslint/consistent-type-definitions": "error",
+      "@typescript-eslint/dot-notation": "error",
+      "@typescript-eslint/explicit-member-accessibility": [
+        "error",
+        {
+          accessibility: "no-public",
+        },
+      ],
+      "@typescript-eslint/explicit-module-boundary-types": "off",
+      "@typescript-eslint/indent": [
+        "error",
+        2,
+        {
+          FunctionDeclaration: {
+            parameters: "first",
+          },
+          FunctionExpression: {
+            parameters: "first",
+          },
+          SwitchCase: 1,
+        },
+      ],
+      "@typescript-eslint/member-delimiter-style": [
+        "error",
+        {
+          multiline: {
+            delimiter: "semi",
+            requireLast: true,
+          },
+          singleline: {
+            delimiter: "semi",
+            requireLast: false,
+          },
+        },
+      ],
+      "@typescript-eslint/member-ordering": [
+        "error",
+        {
+          default: [
+            "public-static-field",
+            "protected-static-field",
+            "private-static-field",
+
+            "public-instance-field",
+            "protected-instance-field",
+            "private-instance-field",
+
+            "public-constructor",
+            "protected-constructor",
+            "private-constructor",
+
+            "public-static-method",
+            "protected-static-method",
+            "private-static-method",
+
+            "public-instance-method",
+            "protected-instance-method",
+            "private-instance-method",
+          ],
+        },
+      ],
+      "@typescript-eslint/naming-convention": "off",
+      "@typescript-eslint/no-explicit-any": "off",
+      "@typescript-eslint/no-parameter-properties": "off",
+      "@typescript-eslint/no-unsafe-assignment": "warn",
+      "@typescript-eslint/no-unsafe-call": "warn",
+      "@typescript-eslint/no-unsafe-member-access": "warn",
+      "@typescript-eslint/no-unsafe-return": "warn",
+      "@typescript-eslint/no-unused-expressions": "error",
+      "@typescript-eslint/no-use-before-define": "off",
+      "@typescript-eslint/prefer-for-of": "error",
+      "@typescript-eslint/prefer-function-type": "error",
+      "@typescript-eslint/quotes": [
+        "error",
+        "single",
+        { avoidEscape: true },
+      ],
+      "@typescript-eslint/restrict-template-expressions": [
+        "error",
+        {
+          allowAny: false,
+          allowBoolean: true,
+          allowNullish: true,
+          allowNumber: true,
+        },
+      ],
+      "@typescript-eslint/semi": [
+        "error",
+        "always",
+      ],
+      "@typescript-eslint/type-annotation-spacing": "error",
+      "@typescript-eslint/unbound-method": "warn", // TODO: some calls are intentionally unbounded...
+      "@typescript-eslint/unified-signatures": "error",
+      "no-fallthrough": "off",
+      "no-invalid-this": "off",
```

Comment:
can use some comment on why rules are turned off (whether it's permanent or a todo)